// Copyright 2020 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// These are protos for driving the XSurface data store.

syntax = "proto2";

package feedxsurface;

option optimize_for = LITE_RUNTIME;

message OfflineBadgeContent {
  optional bool available_offline = 1;
}

message WebFeedFollowState {
  enum FollowState {
    UNSPECIFIED = 0;
    // The WebFeed is followed.
    FOLLOWED = 1;
    // The WebFeed is not followed.
    NOT_FOLLOWED = 2;
    // The client is in the process of trying to follow the WebFeed.
    FOLLOW_IN_PROGRESS = 3;
    // The client is in the process of trying to unfollow the WebFeed.
    UNFOLLOW_IN_PROGRESS = 4;
  }
  optional FollowState follow_state = 1;
}
